Remove interstitial

tohyung85 opened the pull request:
```
Remove interstitial
- Create games.js file for draggable and droppable functionality
- Ajax call implemented on droppable
- Add row and column data to individual chess square elements
- Add update url path data to individual pieces
- Remove interstitial view on piece selection
- Remove show action on pieces controller and associated tests on Rspec
- Refactor games.js file to call data directly instead of attr('data-row')
- Refactor display_piece code to keep it DRY
```

tohyung85 committed:
Install jquery ui and disable turbolinks

tohyung85 committed:
Remove interstitial
- Create games.js file for draggable and droppable functionality
- Ajax call implemented on droppable
- Add row and column data to individual chess square elements
- Add update url path data to individual pieces
- Remove interstitial view on piece selection
- Remove show action on pieces controller and associated tests on Rspec

tohyung85 committed:
Refactoring
- Fix js file to call data directly instead of attr('data-row')
- Refactor display_piece code to keep it DRY

mscoutermarsh commented on app/helpers/pieces_helper.rb:
```diff
@@ -1,56 +1,21 @@
+      css_class += ' ' + piece.color + '-pawn'
     end
+    content_tag(:p, '', class: css_class, data: { update_url: data })
   end
```
nice refactor! 👍

mscoutermarsh commented on app/helpers/pieces_helper.rb:
```diff
@@ -1,56 +1,21 @@
-      display_black_piece(piece, css_class)
+      css_class += ' ' + piece.color + '-pawn'
     end
+    content_tag(:p, '', class: css_class, data: { update_url: data })
```
great use of data-attributes

mscoutermarsh commented:
Looks good! go ahead and merge when you're ready (has merge conflicts)

tohyung85 committed:
Incorporate checking of valid piece moves and if game is in check to movement of pieces

- Pieces controller now checks if movement is valid and returns a json object
- games.js will reverse the move if move is invalid
- games.js will also check if any king is in check if the move is valid

tohyung85 committed:
Fix merge conflicts. Remove /pieces/show.html.erb

tohyung85 commented:
Hey all, have made modifications to check if the movements using javascript are valid moves and reverse changes if not.
App will also check if the game is in_check? after every move

tohyung85 committed:
Fix error on pawn valid move due to white pieces initial position at top of board
- correct valid_move? method for pawn.rb
- Correct spec files

tohyung85 committed:
Refactor display piece method

mscoutermarsh commented on app/controllers/pieces_controller.rb:
```diff
@@ -1,14 +1,12 @@
-
   def update
     @piece = Piece.find(params[:id])
+    return render json: { update_attempt: 'invalid move' } if @piece.valid_move?(params[:piece][:row_coordinate].to_i, params[:piece][:column_coordinate].to_i) == false
```
Another way to write this.

Instead of 

`if something == false`

you can do

`unless something`

mscoutermarsh commented on app/controllers/pieces_controller.rb:
```diff
@@ -1,14 +1,12 @@
+    return render json: { update_attempt: 'invalid move' } if @piece.valid_move?(params[:piece][:row_coordinate].to_i, params[:piece][:column_coordinate].to_i) == false
     @piece.update_attributes(piece_params)
-    redirect_to game_path(@piece.game)
+    in_check = @piece.game.in_check?.present? ? true : false
```
For this, you don't need the ternary. 

`in_check = @piece.game.in_check?.present?` will give the same result. `present?` returns true or false.

mscoutermarsh commented:
Looks great! Go ahead and merge it whenever you're ready

tohyung85 committed:
Clean up code to improve readability
